sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -673,3 +676,16 @@ def validate_cache_staleness_value(max_integrated_cache_staleness):
     int(max_integrated_cache_staleness)  # Will throw error if data type cant be converted to int
     if max_integrated_cache_staleness <= 0:
         raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be a positive integer.")
+
+
+def stringify_auto_scale(offer):
+    auto_scale_params = None
+    max_throughput = offer.auto_scale_max_throughput
+    increment_percent = offer.auto_scale_increment_percent
+    if max_throughput is not None:
+        auto_scale_params = {"maxThroughput": max_throughput}
+        if increment_percent is not None:
+            auto_scale_params["autoUpgradePolicy"] = {"throughputPolicy": {"incrementPercent": increment_percent}}
+    auto_scale_settings = json.dumps(auto_scale_params)
```

Comment:
What happens if increment percent is provided but not max throughput?  Does the service error? Do we need to validate that?

Reply:
Well you wouldn't have an increment percent without the max throughput being provided as well it wouldn't work if you did that because if the max throughput is not provided then you will not get the option to have an increment percent.

Reply:
Well, you could just set `offer.auto_scale_increment_percent` and not set the other. As written, the SDK would silently ignore this, and simply not send any autoscale information at all. Presumably it should send that info and let the service throw an error.

Reply:
Well we don't set the inputs that way because it will not work as it is not deserialized. You can't do `offer.auto_scale_increment_percent` to set the increment percent you would have to do something like `Offer(auto_scale_max_throughput=5000, auto_upgrade_throughput_increment_percent=0)` if you tried to set increment percent without the max throughput you will get an error and it will not work.